mlir/include/mlir/Dialect/MIOpen/Tuning/GridwiseGemmParams.h

```diff
@@ -834,4 +722,105 @@ class PopulateParams : public PopulateParamsBase {
                               int64_t &gemmCDstPerWrite, int64_t &gridSize);
 };
 
+class PopulateParamsXDL : public PopulateParamsBase {
+private:
+  llvm::SmallVector<InitParamsXDL, 4> initParameters = {
+      // M/block N/block K/block M/wave N/wave
+      {128, 128, 16, 64, 64},
+      {8, 64, 8, 8, 64},
+      {4, 64, 16, 4, 64},
+      {16, 16, 4, 16, 16},
+  };
+  const int64_t waveSize = 64;
+
+  int64_t obtainBlockSize(InitParamsXDL &params, int64_t waveSize) {
+    return waveSize * params.gemmNPerBlock * params.gemmMPerBlock /
+           (params.gemmMPerWave * params.gemmNPerWave);
+  }
+
+  LogicalResult calculateGemmABlockCopyPerformanceParameters(
+      InitParamsXDL *param, ConvolutionContext &ctx, DerivedParams &derived) {
+    int64_t blockSize = obtainBlockSize(*param, waveSize);
+    return calculateInputDerivedParams(param, blockSize, ctx, true, derived);
+  }
+
+  LogicalResult calculateGemmBBlockCopyPerformanceParameters(
+      InitParamsXDL *param, ConvolutionContext &ctx, DerivedParams &derived) {
+    int64_t blockSize = obtainBlockSize(*param, waveSize);
+    return calculateInputDerivedParams(param, blockSize, ctx, false, derived);
+  }
+
+  LogicalResult calculateLdsNumberOfByte(InitParamsXDL *param,
+                                         const ConvolutionContext &ctx,
+                                         DerivedParams gemmADerived,
+                                         DerivedParams gemmBDerived,
+                                         size_t &ldsSize) {
+
+    int64_t threadGemmDataPerRead_GemmM =
+        param->gemmMPerBlock / gemmADerived.clusterLenGemmPos2;
+    int64_t threadGemmDataPerRead_GemmN =
+        param->gemmNPerBlock / gemmBDerived.clusterLenGemmPos2;
+
+    const auto max_lds_align =
+        lcm(gemmADerived.dstDataPerWrite, gemmBDerived.dstDataPerWrite,
+            threadGemmDataPerRead_GemmM, threadGemmDataPerRead_GemmN);
+
+    const auto a_block_space =
+        param->gemmKPerBlock *
+        integer_least_multiple(param->gemmMPerBlock, max_lds_align);
+    const auto b_block_space =
+        param->gemmKPerBlock *
+        integer_least_multiple(param->gemmNPerBlock, max_lds_align);
+
+    ldsSize = 2 * (a_block_space + b_block_space) * sizeof(float);
```

Comment:
@jerryyin for XDLOPS we don't do LDS double buffering.

Reply:
Will get rid of `2 *` in next review.